src/main/java/ca/ciccc/silverBullet/gameBoard/GameScene.java

```diff
@@ -63,54 +57,60 @@ public void boardUpdate(){
     }
 
     public void onKeyPressed(KeyCode key){
-        switch (key) {
-            case Q:
-                if (controllingPlayer == 0){
-                    testFirstPlayer.addAction(PlayerAction.TURN_LEFT);
-                } else{
-                    testSecondPlayer.addAction(PlayerAction.TURN_LEFT);
-                }
-                break;
-            case W:
-                if (controllingPlayer == 0){
-                    testFirstPlayer.addAction(PlayerAction.MOVE);
-                } else{
-                    testSecondPlayer.addAction(PlayerAction.MOVE);
-                }
-                break;
-            case E:
-                if (controllingPlayer == 0){
-                    testFirstPlayer.addAction(PlayerAction.TURN_RIGHT);
-                } else{
-                    testSecondPlayer.addAction(PlayerAction.TURN_RIGHT);
-                }
-                break;
-            case R:
-                if (controllingPlayer == 0){
-                    testFirstPlayer.addAction(PlayerAction.SHOOT);
-                } else{
-                    testSecondPlayer.addAction(PlayerAction.SHOOT);
-                }
-                break;
-            case T:
-                if (controllingPlayer == 0){
-                    testFirstPlayer.addAction(PlayerAction.WAIT);
-                } else{
-                    testSecondPlayer.addAction(PlayerAction.WAIT);
-                }
-                break;
-            case SPACE:
-                currentActionNumber  = 0;
-                controllingPlayer = 0;
-                isExecuting = true;
-                testFirstPlayer.setActionsFull(false);
-                testSecondPlayer.setActionsFull(false);
-
-                break;
+        if(!gameBoard.areAllFull()){
+
+            switch (key) {
+                case Q:
+                    gameBoard.players.get(controllingPlayer).addAction(PlayerAction.TURN_LEFT);
+                    break;
+                case W:
+                    gameBoard.players.get(controllingPlayer).addAction(PlayerAction.MOVE);
+                    break;
+                case E:
+                    gameBoard.players.get(controllingPlayer).addAction(PlayerAction.TURN_RIGHT);
+
+                    break;
+                case R:
+                    gameBoard.players.get(controllingPlayer).addAction(PlayerAction.SHOOT);
+
+                    break;
+                case T:
+                    gameBoard.players.get(controllingPlayer).addAction(PlayerAction.WAIT);
+                    break;
+            }
         }
-        if(testFirstPlayer.isActionsFull()){
+        if(!gameBoard.areAllFull() && gameBoard.players.get(controllingPlayer).isActionsFull()){
             controllingPlayer++;
         }
+        if(key.equals(KeyCode.SPACE)){
```

Comment:
if key was null, this causes NPE.
Basically, code bellow is always better

```
if(KeyCode.SPACE.equals(key)){}
```

Reply:
Thank you, changed.